From: Jiri Kosina <jkosina@suse.cz>
Subject: [PATCH] x86/pti: don't report XenPV as vulnerable
References: bsc#1097551
Patch-mainline: Queued in subsystem maintainer repository
Git-repo: git://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git
Git-commit: 66aa6b5cbc359331fc054e96bb49e9502bc0b1d9

Xen PV domain kernel is not by design affected by meltdown as it's enforcing
split CR3 itself. Let's not report such systems as "Vulnerable" in sysfs (we're
also already forcing PTI to off in X86_HYPER_XEN_PV cases); the
security of the system ultimately depends on presence of mitigation in
Hypervisor, which can't be easily detected from DomU; let's report that.

Reported-and-tested-by: Mike Latimer <mlatimer@suse.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Acked-by: Juergen Gross <jgross@suse.com>
Cc: Borislav Petkov <bp@suse.de>
Link: https://lkml.kernel.org/r/nycvar.YFH.7.76.1806180959080.6203@cbobk.fhfr.pm
---
 arch/x86/kernel/cpu/bugs.c |    4 ++++
 1 file changed, 4 insertions(+)

--- a/arch/x86/kernel/cpu/bugs.c
+++ b/arch/x86/kernel/cpu/bugs.c
@@ -652,6 +652,10 @@ ssize_t cpu_show_common(struct device *d
 		if (boot_cpu_has(X86_FEATURE_KAISER))
 			return sprintf(buf, "Mitigation: PTI\n");
 
+		if (boot_cpu_has(X86_FEATURE_XENPV))
+			return sprintf(buf, "Unknown (XEN PV detected, hypervisor "
+					    "mitigation required)\n");
+
 		break;
 
 	case X86_BUG_SPECTRE_V1:
